### DISCUSSION TITLE
Add background repository migration

### DISCUSSION THREAD

**Discussion by geek981108** [gogs/gogs#6809] [Proposal]
### Describe the feature

Thanks of @unknwon 's [article](https://unknwon.io/posts/220210-write-a-proposal-for-open-source-contributions/)!
This issue quotes from and inspired by article.

> **Background** 
When a user wants to migrate a Git repository from other code hosts to Gogs, the user can put an HTTPS URL of the repository and submit the migration request. The repository migration will then start in the foreground, and the page will not be finishing loading until the migration is done. Then user will be redirected to the home page of migrated repository.

>**Problems** 
There is no visibility into the progress of the repository migration, it often times out if the repository is large and takes long time to finish.
Visiting the home page of the migrating repository gives users 500 errors.
If there is a network jitter or server restart, the migration would fail silently and resulted the repository to be migrated in a malformed state on disk, manual intervention is required to remove the repository and start over.

**Description**
Do the migration in background and has status or progress indicator in home page(e.g. “queued”, “cloning” or “failed” )

### Describe the solution you'd like

>1. Create a new database table acting as queue of repository migration tasks.
>2. When users visit the home page of the migrating repository, a status message is shown depending on the migration is “queued”, “cloning” or “failed” (with error).
>3. On restarts, Gogs needs to pick up unfinished migration tasks and continue.

### Describe alternatives you've considered

From the code, we can see some mirror clone cron tasks. So maybe add a status indicator in the home page to show the status of the cron job first would be great.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by unknwon** [comment]
I've converted this to discussions for reply-in-thread support.

So to proceed with this proposal, we need to lay out an iterative plan to move forward. Could you compile a list of steps we need to take to reach the goal of this proposal?
